### PR TITLE
fix: at_auth : use different IVs for encrypting encryptionPrivateKey and selfEncryptionKey.

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- In at_auth package, remove the usage of  legacy Initialization Vector.
- In enroll approval flow, instead of legacy IV's generate random IV's for encrypting the "defaultEncryptionPrivateKey" and "selfEncryptionKey" in APKAM flow.
- The primary reason for storing the encrypted keys is to share defaultEncryptionPrivateKey and defaultSelfEncryptionKey with the app requesting enrollment access. This process is fully handled by the enroll:approve command. Therefore, remove the defaultEncryptionPrivateKey and defaultSelfEncryptionKey when submitting the initial enrollment request.(enroll:request command).

### CODE REVIEW

**Review by murali-shris — DISMISSED** [review]

**Review by gkc — CHANGES_REQUESTED** [review]
functional tests are failing

**Comment by sitaram-kalluri** [comment]
> functional tests are failing

@gkc : The functional tests are failing because the at_onboarding_cli functional tests runs on the "prod" server of secondary server. The changes related to different IV's are merged to trunk branch, and we are yet to release into canary and prod.

Also, can we have two jobs where functional tests run on trunk version of secondary server and prod version of secondary server? or shall we modify the existing to run only on trunk?

**Comment by sitaram-kalluri** [comment]
> functional tests are failing

The functional tests pass now.

File: packages/at_auth/lib/src/enroll/at_enrollment_impl.dart [hunk 1]
```diff
@@ -116,28 +112,36 @@ class AtEnrollmentImpl implements AtEnrollmentBase {
...
         .result;
 
     String command = 'enroll:approve:${jsonEncode({
           'enrollmentId': enrollmentRequestDecision.enrollmentId,
           'encryptedDefaultEncryptionPrivateKey':
               encryptedDefaultEncryptionPrivateKey,
-          'encryptedDefaultSelfEncryptionKey': encryptedDefaultSelfEncryptionKey
+          AtConstants.apkamEncryptionPrivateKeyIV:
+              base64Encode(encryptionPrivateKeyIV.ivBytes),
+          'encryptedDefaultSelfEncryptionKey':
```
Comment (gkc) [inline packages/at_auth/lib/src/enroll/at_enrollment_impl.dart]: This should be in AtConstants too

Please use AtConstants consistently

**Review by gkc — DISMISSED** [review]

**Review by murali-shris — APPROVED** [review]

**Pull request merged**